### PR TITLE
[Doc rework] api.md file modification

### PR DESCRIPTION
**Pull request by mapedraza**
This PR is the first in the serie of #1329 

This PR modifies api.md file 

Documents to be integrated in:

Current file | Current content | Future file
-- | -- | --
advanced-topics.md | specific topics | api.md
api.md | API description | api.md
expressionLanguage.md | Information about expressions | api.md

Task: 

- [x] Rework API routes in a *Orion api.md way* #1335 (+ https://github.com/telefonicaid/iotagent-node-lib/pull/1332/commits/fd2435f15fc128dc40192a2172b40b0f4fb04415)
- [x] Rework existing topics in api.md (in independent PR) https://github.com/telefonicaid/iotagent-node-lib/pull/1336
- [x] Integrate advanced topics.md (into topics) (in independent PR) #1338 
- [x] Integrate expressionLangue into topics (in independent PR) https://github.com/telefonicaid/iotagent-node-lib/pull/1359
- [x] Additional changes (in independent PR) https://github.com/telefonicaid/iotagent-node-lib/pull/1365

### CODE REVIEW

File: doc/development.md [hunk 1]
```diff
@@ -163,3 +163,59 @@ To ensure consistent Markdown formatting run the following:
 # Use git-bash on Windows
 npm run prettier:text
 ```
+
+# DB Models (from API document)
+
```
Comment (mapedraza) [inline doc/development.md:168]: This new section is created, as draft to contains the relationship between database terms and API terms. This information is not required for API users, but it is useful for developers.

**Comment by fgalan** [comment]
Initial commit on this PR ([fd2435f](https://github.com/telefonicaid/iotagent-node-lib/pull/1332/commits/fd2435f15fc128dc40192a2172b40b0f4fb04415)) reviewed and LGTM. Next pieces of work will come in independent PRs on task/doc-rework.

**Comment by mapedraza** [comment]
As discussed in a separate meeting, it makes sense to merge this PR directly on `master` branch, in order to avoid future conflicts with new features. Thus, i change the destination branch to master

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
We have decided to change base branch from task/doc-rework to master.

**Pull request merged**